_books/ion-1-1/src/modules/local_modules.md

```diff
@@ -0,0 +1,132 @@
+# Local modules
+
+Local modules are lexically scoped.
+They can be referenced immediately following their definition, up until the end of their enclosing scope.
+They can be defined either:
+1. **At the top level of a stream**, in which case the enclosing scope is the stream itself.
+2. **Inside another module**, in which case the enclosing scope is the parent module.
+   The parent module can be a shared or local module.
+
+Local modules always have a symbolic name given at the point of definition, also known as a "binding."
+It is legal for a module binding to "shadow" a module binding in its parent scope by using the same name.
+
+```ion
+$ion::
+(module foo // <-- Top-level module `foo`
+  (macro_table
+    (macro quux () Quux)))
+
+$ion::
+(module bar
+  (module foo // <-- Shadows the top-level module `foo`
+    (macro_table
+      (macro quuz () Quuz)))
+  (macro_table foo::quuz) // <-- Refers to the innermost `foo`
+)
+```
+
+However, it is _not_ legal for a local module to use the same name as a module previously defined in the _same_ scope.
+
+```ion
+$ion::
+(module bar
+  (module foo // <-- First definition of `foo` inside `bar`
+    (macro_table
+      (macro quux () Quux)))
+  (module foo // <-- ERROR: module `foo` already defined in this scope
+    (macro_table
+      (macro quuz () Quuz)))
+  /*...*/
+)
+```
+
+The only exception to this rule is at the top level; stream-level bindings can be redefined.
```

Comment:
In other words, stream level bindings are mutable (in some sense) whereas anything inside of a module is immutable.